sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -474,53 +477,57 @@
             if (workingDirectory != null) {
                 builder.directory(new File(workingDirectory));
             } else {
-                throw logger.logExceptionAsError(new IllegalStateException("A Safe Working directory could not be"
+                throw LOGGER.logExceptionAsError(new IllegalStateException("A Safe Working directory could not be"
                     + " found to execute CLI command from. To mitigate this issue, please refer to the troubleshooting "
                     + " guidelines here at https://aka.ms/azsdk/java/identity/azclicredential/troubleshoot"));
             }
             builder.redirectErrorStream(true);
             Process process = builder.start();
 
-            reader = new BufferedReader(new InputStreamReader(process.getInputStream(), "UTF-8"));
-            String line;
             StringBuilder output = new StringBuilder();
-            while (true) {
-                line = reader.readLine();
-                if (line == null) {
-                    break;
-                }
-                if (line.startsWith(WINDOWS_PROCESS_ERROR_MESSAGE) || line.matches(LINUX_MAC_PROCESS_ERROR_MESSAGE)) {
-                    throw LoggingUtil.logCredentialUnavailableException(logger, options,
-                        new CredentialUnavailableException(
-                            "AzureCliCredential authentication unavailable. Azure CLI not installed."
-                                + "To mitigate this issue, please refer to the troubleshooting guidelines here at "
-                                + "https://aka.ms/azsdk/java/identity/azclicredential/troubleshoot"));
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream(),
```

Comment:
Use try-with-resources to remove a try/finally, also the finally block had a return which doesn't make sense